usaspending_api/references/v2/views/filter_trees/tas_filter_tree.py

```diff
@@ -0,0 +1,94 @@
+from usaspending_api.common.helpers.agency_logic_helpers import cfo_presentation_order
+from usaspending_api.accounts.models import TreasuryAppropriationAccount, FederalAccount
+from usaspending_api.common.helpers.agency_logic_helpers import agency_from_identifiers
+from usaspending_api.references.v2.views.filter_trees.filter_tree import DEFAULT_CHILDREN, Node, FilterTree
+from usaspending_api.references.models import ToptierAgency
+
+
+class TASFilterTree(FilterTree):
+    def toptier_search(self):
+        agency_id_sets = TreasuryAppropriationAccount.objects.values("fr_entity_code", "agency_id").distinct(
+            "agency_id"
+        )
+        agency_set = {
+            ToptierAgency.objects.filter(
+                toptier_code=agency_from_identifiers(elem["agency_id"], elem["fr_entity_code"])
+            ).first()
+            for elem in agency_id_sets
+        }
```

Comment:
The way this is written, it is pinging the database about 320 times to generate the top level of the tree which results in a 30 second response time when I run it against `dev`.  I believe this can be written in a single query which should help.